packages/material-ui-lab/src/Masonry/Masonry.js

```diff
@@ -0,0 +1,182 @@
+// A masonry component using the following libs as inspiration.
+// - https://github.com/STRML/react-grid-layout
+// - https://next.material-ui.com/components/image-list/#masonry-image-list
+// - https://github.com/desandro/masonry
+// - https://w3bits.com/css-grid-masonry
+
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { createUnarySpacing, getValue, handleBreakpoints } from '@material-ui/system';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import { styled, useThemeProps } from '@material-ui/core/styles';
+import { deepmerge } from '@material-ui/utils';
+import { getMasonryUtilityClass } from './masonryClasses';
+import MasonryContext from './MasonryContext';
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getMasonryUtilityClass, classes);
+};
+
+// Duplicated with Stack.js
+function resolveBreakpointValues({ values, base }) {
```

Comment:
We could consider extracting this and reusing it in both components. Maybe we could add it as a utility in the @material-ui/system.